server/src/db.ts

```diff
@@ -70,3 +91,57 @@ export async function editZone(zone: Zone): Promise<void> {
 export async function deleteZone(id: number): Promise<void> {
   await dbRun('DELETE FROM zones WHERE id = ?', [id]);
 }
+
+// ---- Light Operations ----
+
+export async function getLights(): Promise<Light[]> {
+  return dbAll(`SELECT * FROM lights`) as Promise<Light[]>;
+}
+
+export async function createLight(
+  lightRequest: CreateLightRequest
+): Promise<void> {
+  switch (lightRequest.type) {
+    case LightType.RVL: {
+      const rvlLightRequest: CreateRVLLightRequest = lightRequest as CreateRVLLightRequest;
+      await dbRun(`INSERT INTO lights (name, type, channel) values (?, ?, ?)`, [
+        rvlLightRequest.name,
+        LightType.RVL,
+        rvlLightRequest.channel
+      ]);
+      break;
+    }
+    case LightType.Hue: {
+      const hueLightRequest: CreateHueLightRequest = lightRequest as CreateHueLightRequest;
+      await dbRun(`INSERT INTO lights (name, type) values (?, ?)`, [
+        hueLightRequest.name,
+        LightType.RVL
+      ]);
+      break;
+    }
+  }
+}
+
+export async function editLight(light: Light): Promise<void> {
+  switch (light.type) {
+    case LightType.RVL:
+      const rvlLight: RVLLight = light as RVLLight;
+      await dbRun('UPDATE lights SET name = ?, channel = ? WHERE id = ?', [
+        rvlLight.name,
+        rvlLight.channel,
```

Comment:
Should we be doing any validation on the channel?

Reply:
Yeah sure, will do.